### PR TITLE
[python wrappers] Add user config and data dirs

### PR DESCRIPTION
**Pull request by Cjkjvfnby**
Add the config directory and option access. I need this code to have the way to generate AI stubs for that interfaces like it is done for  [AI](https://github.com/freeorion/freeorion/blob/master/default/python/AI/freeOrionAIInterface.pyi#L1)

I know a little about C++ and don't compile code at all. 

@o01eg  can you take care of this PR?

### CODE REVIEW

**Comment by o01eg** [comment]
Does it need additional methods in both `ServerWrapper` and `UniverseWrapper`? I suppose only one should be enough.

File: python/UniverseWrapper.cpp [hunk 1]
```diff
@@ -239,6 +241,23 @@ namespace {
     { return rules.RuleExists(name); }
     bool RuleExistsWithType(const GameRules& rules, const std::string& name, GameRules::Type type)
     { return rules.RuleExists(name, type); }
+
+    boost::python::object GetOptionsDBOptionStr(std::string const &option)
```
Comment (geoffthemedio) [inline python/UniverseWrapper.cpp]: `std::string const &option` -> `const std::string& option`

Comment (Cjkjvfnby) [reply python/UniverseWrapper.cpp]: It is copypaste from AIWrapper

**Comment by Cjkjvfnby** [comment]
> Does it need additional methods in both ServerWrapper and UniverseWrapper? I suppose only one should be enough.

I Checked usages in turn events and universe generation, looks like both uses methods defined in servers.

File: python/server/ServerWrapper.cpp [hunk 2]
```diff
@@ -1392,4 +1409,15 @@ namespace FreeOrionPython {
         def("planet_make_colony",                   PlanetMakeColony);
         def("planet_cardinal_suffix",               PlanetCardinalSuffix);
     }
+
+        /////////////////
```
Comment (o01eg) [inline python/server/ServerWrapper.cpp:1413]: The code doesn't work outside functions.

File: python/server/ServerWrapper.cpp [hunk 3]
```diff
@@ -1392,4 +1409,15 @@ namespace FreeOrionPython {
...
+        //   Configs   //
+        /////////////////
+        def("get_options_db_option_str",    GetOptionsDBOptionStr,     return_value_policy<return_by_value>(), "Returns the string value of option in OptionsDB or None if the option does not exist.");
+        def("get_options_db_option_int",    GetOptionsDBOptionInt,     return_value_policy<return_by_value>(), "Returns the integer value of option in OptionsDB or None if the option does not exist.");
+        def("get_options_db_option_bool",   GetOptionsDBOptionBool,    return_value_policy<return_by_value>(), "Returns the bool value of option in OptionsDB or None if the option does not exist.");
+        def("get_options_db_option_double", GetOptionsDBOptionDouble,  return_value_policy<return_by_value>(), "Returns the double value of option in OptionsDB or None if the option does not exist.");
+
+        def("get_user_config_dir",         GetUserConfigDirWrapper,        /* no return value policy, */ "Returns path to directory where FreeOrion stores user specific configuration.");
```
Comment (o01eg) [inline python/server/ServerWrapper.cpp:1421]: It's already defined in `WrapServer`

**Review by Vezzra — CHANGES_REQUESTED** [review]
In their current form, the proposed changes won't work. I've provided some comments about what to change, I'm also going to provide a PR for your branch where you can see how I'd change things.

**Comment by Morlic-fo** [comment]
Is there a reason to copy-paste the AIWrapper code to the ServerWrapper rather than moving it into a common source file?

**Comment by Vezzra** [comment]
I wasn't aware that this wrapper code has been copied from the AIWrapper. If that's the case, then yes, instead of duplicating the code this should definitely be moved into the common wrappers code.

There are already existing source files for common wrappers, so that's where this should be moved to.

**Comment by Cjkjvfnby** [comment]
Hi,   thanks for collaborating. 

Looks like I should not code C++. (especially in notepad).   I am not common with project structure too. 

I am not compiling, so I cannot test it now. I use Monday builds.

Feel free to push to my repo or abandon this branch and push to master in other way (Looks like nothing from my code is left).

**Comment by o01eg** [comment]
@Cjkjvfnby Are you going to close this PR or accept changes from @Vezzra or me?

**Comment by Vezzra** [comment]
> @Cjkjvfnby Are you going to close this PR or accept changes from @Vezzra or me?

Actually both this PR and my proposed changes should be discarded, and @Morlic-fo's suggestion implemented instead (move the functions copied over from the AIWrapper to the common wrappers), to avoid code duplication.

**Comment by Vezzra** [comment]
@Cjkjvfnby, @o01eg: I have provided a PR which implements moving the wrappers from the AI to the common wrappers instead of duplicating them for the server: #2271.

**Comment by Dilvish-fo** [comment]
I don' think we need to wait for the other PR to be completed/accepted before closing this one as superseded.

**Pull request closed without merging**